keyserver/storage/sqlite3/device_keys_table.go

```diff
@@ -52,6 +52,9 @@ const selectDeviceKeysSQL = "" +
 const selectBatchDeviceKeysSQL = "" +
 	"SELECT device_id, key_json, stream_id, display_name FROM keyserver_device_keys WHERE user_id=$1 AND key_json <> ''"
 
+const selectBatchDeviceKeysWithEmptiesSQL = "" +
+	"SELECT device_id, key_json, stream_id, display_name FROM keyserver_device_keys WHERE user_id=$1"
```

Comment:
Given `key_json TEXT NOT NULL` why do we need this and why is `selectBatchDeviceKeysSQL` not suitable?

Reply:
It's `NOT NULL` but that doesn't prevent it from being `EMPTY` which apparently it is when the device is first created (as earlier comment). I agree it's a weird shape but it's pre-existing.

Reply:
Yeah I think I remember why now - deleted devices need to be propogated over federation and hence need a stream ID etc, so an empty device is the tombstone.